### PR TITLE
Add diagnostics(mfg) interface in wpantund

### PR DESCRIPTION
**Pull request by chshu**
- Spinel: add 'SPINEL_PROP_NEST_STREAM_MFG' property

  - D-Bus: add 'WPANTUND_IF_CMD_MFG' command

  - Move original 'NCPMfgInterface' to 'NCPMfgInterface_v0', add new 'NCPMfgInterface_v1'

### CODE REVIEW

**Comment by darconeous** [comment]
Why not use `SpinelNCPTaskSendCommand` instead of writing a new task named `SpinelNCPTaskMfg`? The invocation would look something like this:

```
mNCPInstance->start_new_task(boost::shared_ptr<SpinelNCPTask>(
	new SpinelNCPTaskSendCommand(
		mNCPInstance,
		cb,
		SpinelPackData(
			SPINEL_FRAME_PACK_CMD_PROP_VALUE_SET(SPINEL_DATATYPE_UTF8_S),
			SPINEL_PROP_NEST_STREAM_MFG,
			mfg_command.c_str()
		),
		SPINEL_DATATYPE_UTF8_S
	)
));
```

File: src/ncp-spinel/SpinelNCPControlInterface.cpp [hunk 1]
```diff
@@ -438,6 +439,20 @@ SpinelNCPControlInterface::get_name() {
 	return mNCPInstance->get_name();
 }
 
+void
+SpinelNCPControlInterface::mfg(
+    const std::string& mfg_command,
+    CallbackWithStatusArg1 cb
+) {
+	mNCPInstance->start_new_task(boost::shared_ptr<SpinelNCPTask>(
+		new SpinelNCPTaskMfg(
```
Comment (darconeous) [inline src/ncp-spinel/SpinelNCPControlInterface.cpp]: Do this instead:

```
mNCPInstance->start_new_task(boost::shared_ptr<SpinelNCPTask>(
    new SpinelNCPTaskSendCommand(
        mNCPInstance,
        cb,
        SpinelPackData(
            SPINEL_FRAME_PACK_CMD_PROP_VALUE_SET(SPINEL_DATATYPE_UTF8_S),
            SPINEL_PROP_NEST_STREAM_MFG,
            mfg_command.c_str()
        ),
        SPINEL_DATATYPE_UTF8_S
    )
));
```

Unless `SpinelNCPTaskMfg` is doing something special that I'm not realizing.

**Comment by darconeous** [comment]
You are going to need to rename `NCPMfgInterface` class if you are going to introduce v0 and v1 versions.

File: src/ncp-spinel/SpinelNCPControlInterface.cpp (hunk 1 above)
Comment (chshu) [inline src/ncp-spinel/SpinelNCPControlInterface.cpp]: I have removed `SpinelNCPTaskMfg`, and use `SpinelNCPTaskSendCommand` instead.
My original idea is to make `mfg` a separated feature as in previous v0 implementation, but it seems unnecessary now.

**Comment by chshu** [comment]
@darconeous , Thanks for your suggestions, I have updated the implementation, please have a review again, thanks.

**Comment by darconeous** [comment]
I'm good with this. I want to get @abtink's thoughts before merging.

**Comment by abtink** [comment]
LGTM :+1:

**Pull request merged**